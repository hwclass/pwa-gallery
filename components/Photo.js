import React from 'react'
import Layout from './Layout';

export default class extends React.Component {
  render() {
    const { id, uri, title } = this.props
    return (
      <Layout title={`${process.env.TITLE} - ${title}`} headerText={`${process.env.PHOTO_TITLE}`}>
        <article>
          <span className="title">
            {title}
            <style jsx>{`
              .title {
                display: block;
                max-width: 100%;
                padding: 15px;
                text-align: center;
                font-size: 2.5vmin;
                background: #000;
                color: #FFF;
                white-space: nowrap;
                overflow: hidden;
                text-overflow: ellipsis;
              }
              @media screen and (max-width: 400px) {
                .title {
                  font-size: 2.5vmax;
                }
              }
            `}</style>
          </span>
          <picture key={id} className="pic">
            <img src={`http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg`}/>
            <style jsx>{`
              .pic {
                background: #FFF;
              }
              img {
                display: block;
                margin: auto;
                width: 100%;
              }
              @media screen and (max-width: 400px) {
                img {
                  width: 100%;
                }
              }
            `}</style>
          </picture>
        </article>
      </Layout>
    );
  }
}
