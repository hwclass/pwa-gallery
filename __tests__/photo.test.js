/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Photo from '../components/Photo'

describe('Photo', () => {
  it('should render', () => {
    const component = shallow(<Photo id='1' uri='/some-url/for-image-1.jpg' title='Photo Title'/>)
    expect(component.getElements()).toMatchSnapshot();
  })
})

describe('Photo', () => {
  it('should match with snapshot', () => {
    const component = renderer.create(<Photo id='1' uri='/some-url/for-image-1.jpg' title='Photo Title'/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})