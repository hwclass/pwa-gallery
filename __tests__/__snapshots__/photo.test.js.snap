// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should match with snapshot 1`] = `
<div
  className="jsx-3488773145"
>
  <header
    className="jsx-1922746274"
  >
    Photo
  </header>
  <article>
    <span
      className="jsx-588413163 title"
    >
      Photo Title
    </span>
    <picture
      className="jsx-2627651806 pic"
    >
      <img
        className="jsx-2627651806"
        src="http:///some-url/for-image-1.jpg_27.jpg"
      />
    </picture>
  </article>
</div>
`;

exports[`Photo should render 1`] = `
Array [
  <_class
    headerText="Photo"
    title="PWA Gallery - Photo Title"
  >
    <article>
      <span
        className="jsx-588413163 title"
      >
        Photo Title
        <JSXStyle
          css=".title.jsx-588413163{display:block;max-width:100%;padding:15px;text-align:center;font-size:2.5vmin;background:#000;color:#FFF;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;}@media screen and (max-width:400px){.title.jsx-588413163{font-size:2.5vmax;}}"
          styleId="588413163"
        />
      </span>
      <picture
        className="jsx-2627651806 pic"
      >
        <img
          className="jsx-2627651806"
          src="http:///some-url/for-image-1.jpg_27.jpg"
        />
        <JSXStyle
          css=".pic.jsx-2627651806{background:#FFF;}img.jsx-2627651806{display:block;margin:auto;width:100%;}@media screen and (max-width:400px){img.jsx-2627651806{width:100%;}}"
          styleId="2627651806"
        />
      </picture>
    </article>
  </_class>,
]
`;
